### PR TITLE
task 5 add authentication and delete course ability

### PR DESCRIPTION
**Pull request by filipHartman**
The functionality is provided but still need to write some of the tests

### CODE REVIEW

File: src/app/pages/courses/courses.module.ts [hunk 1]
```diff
@@ -20,6 +22,8 @@ import { CoursesComponent } from './courses.page';
     PipesModule,
     ReactiveFormsModule,
     DirectivesModule,
+    ModalCardModule,
+    MatDialogModule,
```
Comment (dragonflypl) [inline src/app/pages/courses/courses.module.ts:26]: Good job on modules

File: src/app/pages/courses/courses.page.ts [hunk 2]
```diff
@@ -19,7 +21,10 @@ export class CoursesComponent {
 
   searchControl = new FormControl('');
 
-  constructor(private readonly coursesService: CoursesService) {}
+  constructor(
+    private readonly coursesService: CoursesService,
+    public dialog: MatDialog,
```
Comment (dragonflypl) [inline src/app/pages/courses/courses.page.ts:26]: could be readonly as well ;)

File: src/app/pages/login/login.page.spec.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        {
+          provide: Router,
+          useValue: routerMock,
+        },
+      ],
+    });
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
Comment (dragonflypl) [inline src/app/pages/login/login.page.spec.ts:23]: Well tests could be better, couldn't they? :)

File: src/app/pages/login/login.page.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  loginForm: FormGroup = this.createLoginForm();
+
+  constructor(
+    private readonly fb: FormBuilder,
+    private readonly auth: AuthService,
+    private readonly router: Router,
+  ) {}
+
+  ngAfterViewInit(): void {
+    this.subscription = fromEvent(this.submit.nativeElement, 'click').subscribe(
```
Comment (dragonflypl) [inline src/app/pages/login/login.page.ts:48]: why `fromEvent` and not `submit` event from the form? with `submit` event you get handling of `enter`  key, you don't have to unsubscribe manually etc...

File: src/app/shared/components/modal-card/modal-card.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  it('should create modal component', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should call close method of modal ref without anything after clicking no button', () => {
+    const { getByText, click } = component;
+    const modalRef = TestBed.inject(MatDialogRef);
+    spyOn(modalRef, 'close');
+    click(getByText('No'));
+    expect(modalRef.close).toHaveBeenCalled();
```
Comment (dragonflypl) [inline src/app/shared/components/modal-card/modal-card.component.spec.ts:35]: Nice tests

File: src/app/shared/services/auth/auth.service.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { User } from '@interfaces/user';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  private readonly KEY = 'user';
+
+  login(user: User): void {
```
Comment (dragonflypl) [inline src/app/shared/services/auth/auth.service.ts:10]: Although ok, you should use reactive approach:

https://blog.angular-university.io/how-to-build-angular2-apps-using-rxjs-observable-data-services-pitfalls-to-avoid/
with behaviour subject or sth like this:

https://ngrx.io/guide/component-store

i've discovered it recently

**Review by dragonflypl — APPROVED** [review]

File: src/app/pages/login/login.page.spec.ts (hunk 3 above)
Comment (filipHartman) [reply src/app/pages/login/login.page.spec.ts:23]: Yes, I totally agree! I wanted to create a PR by the end of Sunday and skipped some tests. I will provide more testing in the next PR.

File: src/app/pages/login/login.page.ts (hunk 4 above)
Comment (filipHartman) [reply src/app/pages/login/login.page.ts:48]: Thanks for the advice, I will change it to submit

File: src/app/shared/services/auth/auth.service.ts (hunk 6 above)
Comment (filipHartman) [reply src/app/shared/services/auth/auth.service.ts:10]: About the reactive approach, I planned to make it reactive during HTTP module task.

File: src/app/pages/courses/courses.page.ts (hunk 2 above)
Comment (filipHartman) [reply src/app/pages/courses/courses.page.ts:26]: Didn't notice thanks

**Pull request merged**